lib/Extension/LanguageServerPsalm/Model/DiagnosticsParser.php

```diff
@@ -39,23 +64,27 @@ public function parse(string $jsonString, string $filename): array
     }
 
     /**
-     * @return array<mixed>
+     * @return array<PsalmDiagnostic>
      */
     private function decodeJson(string $jsonString): array
     {
-        $decoded = json_decode($jsonString, true, JSON_THROW_ON_ERROR);
```

Comment:
This was wrong previously. Using `JSON_THROW_ON_ERROR` here just sets the recursion limit to some very very big limit.